webapp/src/dogma/features/project/settings/mirrors/DeleteMirror.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  Button,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { newNotification } from 'dogma/features/notification/notificationSlice';
+import ErrorMessageParser from 'dogma/features/services/ErrorMessageParser';
+import { useAppDispatch } from 'dogma/hooks';
+import { MdDelete } from 'react-icons/md';
+
+export const DeleteMirror = ({
+  projectName,
+  id,
+  deleteMirror,
+  isLoading,
+}: {
+  projectName: string;
+  id: string;
+  deleteMirror: (projectName: string, id: string) => Promise<void>;
+  isLoading: boolean;
+}): JSX.Element => {
+  const { isOpen, onToggle, onClose } = useDisclosure();
+  const dispatch = useAppDispatch();
+  const handleDelete = async () => {
+    try {
+      await deleteMirror(projectName, id);
+      dispatch(newNotification('Mirror deleted.', `Successfully deleted ${id}`, 'success'));
+      onClose();
+    } catch (error) {
+      dispatch(newNotification(`Failed to delete ${id}`, ErrorMessageParser.parse(error), 'error'));
+    }
+  };
+  return (
+    <>
+      <Button leftIcon={<MdDelete />} colorScheme="red" size="sm" variant="ghost" onClick={onToggle}>
+        Delete
+      </Button>
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Are you sure?</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            Delete mirror &apos{id}&apos from {projectName}?
+          </ModalBody>
+          <ModalFooter>
+            <HStack spacing={3}>
+              <Button colorScheme="red" variant="outline" onClick={onClose}>
+                Cancel
+              </Button>
+              <Button colorScheme="red" onClick={handleDelete} isLoading={isLoading} loadingText="Deleting">
+                Delete
+              </Button>
+            </HStack>
+          </ModalFooter>
+        </ModalContent>
+      </Modal>
```

Comment:
How about extract this block as a component and sharing it with `DeleteCredential`?